website/upgrade-guides/0-14.html.markdown

```diff
@@ -0,0 +1,303 @@
+behavior that you may need to consider when upgrading. This guide is intended
+to help with that process.
+
+The goal of this guide is to cover the most common upgrade concerns and
+issues that would benefit from more explanation and background. The exhaustive
+list of changes will always be
+[the Terraform Changelog](https://github.com/hashicorp/terraform/blob/v0.14/CHANGELOG.md).
+After reviewing this guide, we recommend reviewing the Changelog to check for
+specific notes about less-commonly-used features. This guide is also not
+intended as an overview of the new features in Terraform v0.14. This release
+includes other enhancements that don't need any special attention during
+upgrade, but those are described in the changelog and elsewhere in the
+Terraform documentation.
+
+This guide focuses on changes from v0.13 to v0.14. Terraform supports upgrade
+tools and features only for one major release upgrade at a time, so if you are
+currently using a version of Terraform prior to v0.13 please upgrade through
+the latest minor releases of all of the intermediate versions first, reviewing
+the previous upgrade guides for any considerations that may be relevant to you.
+
+In particular, Terraform v0.14 no longer includes the `terraform 0.13upgrade`
+command for automatically inserting
+[provider requirements](/docs/configuration/provider-requirements.html)
+into existing modules, and the automatic mechanisms to upgrade legacy provider
+references in the Terraform state. You will need to successfully complete a
+`terraform apply` at least once under Terraform v0.13 before upgrading an
+existing configuration to Terraform v0.14.
+
+-> If you run into any problems during upgrading that are not addressed by the
+information in this guide, please feel free to start a topic in
+[The Terraform community forum](https://discuss.hashicorp.com/c/terraform-core),
+describing the problem you've encountered in enough detail that other readers
+may be able to reproduce it and offer advice.
+
+Upgrade guide sections:
+
+* [Before You Upgrade](#before-you-upgrade)
+* [Provider Dependency Lock File](#provider-dependency-lock-file)
+* [Concise Terraform Plan Output](#concise-terraform-plan-output)
+* [Sensitive Values in Plan Output](#sensitive-values-in-plan-output)
+
+## Before You Upgrade
+
+When upgrading between major releases, we always recommend ensuring that you
+can run `terraform plan` and see no proposed changes on the previous version
+first, because otherwise pending changes can add additional unknowns into the
+upgrade process.
+
+For this upgrade in particular, you must also have successfully run
+`terraform apply` at least once with Terraform v0.13 in order to complete an
+upgrade to the state snapshot format. We recommend doing that with no other
+changes pending.
+
+## Provider Dependency Lock File
+
+In Terraform v0.13 and earlier, the `terraform init` command would always
```

Comment:
I will let you decide if this matters - you wrote it this way on purpose so I won't argue further - but I worry that users might read this and misunderstand since it's (v) _mildly_ inaccurate: 

Technically the _first_ init does this; subsequent `init`s don't install the newest version without `-upgrade`. 

Incredibly nitpicky? Absolutely, please go ahead and ignore it. I just worry that this is just misleading enough to confuse folks.

Reply:
For the sake of this guide I'm basically ignoring the possibility of running `terraform init` again against an already-initialized working directory, because this feature is about teams working together on a configuration where they would not be sharing a working directory and therefore one person's `terraform init` would not affect another person's.

Having thought about this again with fresh brain today, I think still stand by that; the current behavior isn't an effective way to keep your dependencies pinned because in practice working directories rarely live long enough for the temporary files cached in `.terraform` to persist between two operations. I think mentioning that here would only serve to complicate matters by drawing in a bunch of additional concepts that aren't needed to make the decisions this section is calling for readers to make.